### ISSUE TITLE
Issues with merging data from multiple coverage databases (<foo> does not match <bar>)

### ISSUE THREAD

**Issue by nthykier** [pjcj/Devel--Cover#151]
Hi,

The setup is the same as in #150.  Though, now that I got coverage from more test cases, I am starting to see:

```
<configure_output> does not match <__ANON__> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<__ANON__> does not match <setup_work_pool> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<setup_work_pool> does not match <load_profile_and_configure_tags> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<load_profile_and_configure_tags> does not match <load_and_select_collections> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<load_and_select_collections> does not match <__ANON__> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<__ANON__> does not match <parse_options> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<parse_options> does not match <_update_profile> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<_update_profile> does not match <timed_task> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<timed_task> does not match <interrupted> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
<interrupted> does not match <END> - using later value at /usr/lib/x86_64-linux-gnu/perl5/5.22/Devel/Cover/DB.pm line 312.
```

I have made the following observations:
- I note that all of these are subroutine names from "[lintian](https://anonscm.debian.org/cgit/lintian/lintian.git/tree/commands/lintian)".
  -  It is called (exec'ed) via the "[dplint](https://anonscm.debian.org/cgit/lintian/lintian.git/tree/frontend/dplint)" command (which basically prepares some ENV to setup @INC + PERL5OPTS)
- Only some coverage databases causes this issue.  In fact, the majority does not cause these issues.
- Most (but not all) of the subroutines mentioned are involved in GetOptions (i.e. in a "key => subroutine" hash).

**Comment by pink-mist** [comment]
I get the same issue when I try merging databases from testing on perl ```5.16``` with a database generated by testing on perl ```5.18``` or ```5.22```.

**Comment by nthykier** [comment]
I would be surprised if results from Perl 5.X is compatible with Perl 5.Y given perl changes its op-code every now and then.

Anyway, to avoid any doubt: In the the issue I reported, the same version of perl was used for all runs.

**Comment by pjcj** [comment]
The Devel::Cover databases should be compatible across Perl versions.  And across Devel::Cover versions which use the same DB version.  When that changes Devel::Cover should behave appropriately.

I suspect some problem related running different  string evals in different runs.  The lazy loading you mentioned in #150 would fall under that category so I think you may be correct there.

Unfortunately, this is quite a tricky area to get right.

Thanks very much for your report and I hope to investigate this in more detail shortly.

**Comment by pjcj** [comment]
I believe this was fixed with #199 so I am going to close it.  It that is not the case, please feel free to reopen it.

Thanks!
